### ISSUE TITLE
Update detekt to latest version

### ISSUE THREAD

**Issue by vanlooverenkoen** [icapps/android_gradle_build#23]
Update the gradle plugin to the latest detekt version.
Also change the config that can be used in detekt.

requested full config:

```
detekt {
    version = "detekt-version-to-use"

    config = "$projectDir/../codecheck/detekt-config.yml"
    input = "$projectDir/src/main/kotlin"
    filters = ".*test.*,.*/resources/.*,.*/tmp/.*"
    outputName = "detektReport"
    output = "$projectDir/reports/"
}
```

**Comment by vanlooverenkoen** [comment]
Detekt Check Github: https://github.com/arturbosch/detekt
Gradle plugin source code: https://github.com/arturbosch/detekt/tree/master/detekt-gradle-plugin

**Comment by vanlooverenkoen** [comment]
Gradle version 4.9+ will be required for the new detekt plugin

**Comment by vanlooverenkoen** [comment]
Should be fixed in version 1.1.0
